### PR TITLE
Inject Dao's in activities/fragments using Dagger

### LINKED ISSUE #115: Remove singleton pattern from Application class
`Share` application class stores a static reference to the application's context. We should get rid of it and use the context available to us via fragment/activity

### PR DESCRIPTION
**Pull request by shobhitagarwal1612**
Closes #115 

 - Removes static application's instance and it's usage from everywhere
 - Makes activities/fragments injectable
 - Let dagger handle injection for Dao instead of creating new instances manually

### CODE REVIEW

**Comment by shobhitagarwal1612** [comment]
@lakshyagupta21 please review

File: share_app/src/main/java/org/odk/share/adapters/FormsAdapter.java [hunk 1]
```diff
@@ -29,16 +29,20 @@
 
 public class FormsAdapter extends CursorRecyclerViewAdapter<FormsAdapter.FormHolder> {
 
+    private final InstancesDao instancesDao;
+    private final TransferDao transferDao;
     @Nullable
-    private LinkedHashSet<Long> selectedForms;
+    private final LinkedHashSet<Long> selectedForms;
```
Comment (lakshyagupta21) [inline share_app/src/main/java/org/odk/share/adapters/FormsAdapter.java]: Any reason for making it final?

**Comment by lakshyagupta21** [comment]
And there are some merge conflicts as well.

File: share_app/src/main/java/org/odk/share/adapters/FormsAdapter.java (hunk 1 above)
Comment (shobhitagarwal1612) [reply share_app/src/main/java/org/odk/share/adapters/FormsAdapter.java]: To ensure that the variable is never mutated again in the class.

**Comment by shobhitagarwal1612** [comment]
Fixed

**Pull request merged**